tcs/csp_solver_core.cpp

```diff
@@ -1689,966 +1718,534 @@ void C_csp_solver::Ssimulate(C_csp_solver::S_sim_setup & sim_setup)
-				mc_pc_htf_state_in.m_temp = m_cycle_T_htf_hot_des - 273.15;	//[C]
-				mc_pc_htf_state_in.m_pres = m_cycle_P_hot_des;		//[kPa]
-				mc_pc_htf_state_in.m_qual = m_cycle_x_hot_des;		//[-]
-				mc_pc_inputs.m_m_dot = 0.0;		//[kg/hr] no mass flow rate to power cycle
-					// Inputs
-				mc_pc_inputs.m_standby_control = C_csp_power_cycle::OFF;
-					// Performance Call
-				mc_power_cycle.call(mc_weather.ms_outputs,
-					mc_pc_htf_state_in,
-					mc_pc_inputs,
-					mc_pc_out_solver,
-					mc_kernel.mc_sim_info);
-
-				if( m_is_tes )
-				{
-					mc_tes.idle(mc_kernel.mc_sim_info.ms_ts.m_step, mc_weather.ms_outputs.m_tdry + 273.15, mc_tes_outputs);
-
-
-					// If not actually charging (i.e. mass flow rate = 0.0), what should the temperatures be?
-					mc_tes_ch_htf_state.m_m_dot = 0.0;										//[kg/hr]
-					mc_tes_ch_htf_state.m_temp_in = mc_tes_outputs.m_T_hot_ave - 273.15;	//[C] convert from K
-					mc_tes_ch_htf_state.m_temp_out = mc_tes_outputs.m_T_cold_ave - 273.15;	//[C] convert from K
-
-					// If not actually discharging (i.e. mass flow rate = 0.0), what should the temperatures be?
-					mc_tes_dc_htf_state.m_m_dot = 0.0;										//[kg/hr]
-					mc_tes_dc_htf_state.m_temp_in = mc_tes_outputs.m_T_cold_ave - 273.15;	//[C] convert from K
-					mc_tes_dc_htf_state.m_temp_out = mc_tes_outputs.m_T_hot_ave - 273.15;	//[C] convert from K
-				}
-
-				are_models_converged = true;
-
-				break;		// exit switch() after CR_OFF__PC_OFF__TES_OFF__AUX_OFF:
-
-			case CR_OFF__PC_SU__TES_DC__AUX_OFF:
-			{
-				// Use thermal storage to startup power cycle
-				// This solver iterates to find the thermal storage outlet temperature to the power cycle
-				//    and the power cycle demand mass flow rate that reach system equilibrium
-
-				if( !mc_collector_receiver.m_is_sensible_htf )
-				{
-					std::string err_msg = util::format("Operating mode, %d, is not configured for DSG mode", operating_mode);
-					throw(C_csp_exception(err_msg, "CSP Solver"));
-				}
-
-				// Set Solved Controller Variables Here (that won't be reset in this operating mode)
-				m_defocus = 1.0;
-
-				double step_tol = step_tolerance;		//[s]
-				double step_pc_su = std::numeric_limits<double>::quiet_NaN();
-
-				int exit_mode = CSP_CONVERGED;
-				double T_pc_in_exit_tolerance = std::numeric_limits<double>::quiet_NaN();
+                if (!mc_collector_receiver.m_is_sensible_htf)
+                {
+                    std::string err_msg = util::format("Operating mode, %d, is not configured for DSG mode", operating_mode);
+                    throw(C_csp_exception(err_msg, "CSP Solver"));
+                }
 
-				solver_pc_su_controlled__tes_dc(step_tol, 
-					step_pc_su,
-					exit_mode, T_pc_in_exit_tolerance);
+                int cr_mode = C_csp_collector_receiver::ON;
+                int pc_mode = C_csp_power_cycle::OFF;
+                int solver_mode = C_MEQ__m_dot_tes::E__CR_OUT__0;
+                int step_target_mode = C_MEQ__timestep::E_STEP_FIXED;
```

Comment:
These variables should not be ints but rather the enumeration type being used to set their values.

Reply:
Controller-scope variables fixed here https://github.com/NREL/ssc/commit/ee3729d6ed0582d00a9900756fb9e7d24b845e7d

Can add downstream component enumerations to Issues